standard/navbar.lua

```diff
@@ -1,122 +1,156 @@
-			:tag(args.mini and 'abbr' or 'span')
-				:attr('title', 'View this template')
-				:cssText(args.fontstyle)
-				:wikitext(args.mini and 'v' or 'view')
+	local shortcutObjects = {
+		{long = 'view', short = 'v', text = 'View this template', link = templateTitle.fullText},
+		{long = 'talk', short = 'd', text = 'Discuss this template', link = talkpage},
+	}
+	if not Logic.readBool(args.noedit) then
+		table.insert(
+			shortcutObjects,
+			{
+				long = 'edit',
+				short = 'e',
+				text = 'Edit this template',
+				link = templateTitle:fullUrl('action=edit'),
+				externalLink = true
+			}
+		)
+	end
+
+	local shortcuts = mw.html.create('ul')
+		:addClass('hlist')
+
+	for _, item in ipairs(shortcutObjects) do
+		shortcuts:tag('li')
+			:addClass('nv-' .. item.long)
+			:wikitext(item.externalLink and '[' or '[[')
+			:wikitext(item.link .. '|')
+			:tag(isMini and 'abbr' or 'span')
+				:attr('title', item.text)
+				:cssText(fontStyle)
+				:wikitext(isMini and item.short or item.long)
 				:done()
-			:wikitext(']]')
+			:wikitext(item.externalLink and ']' or ']]')
 			:done()
-		:tag('li')
-			:addClass('nv-talk')
-			:wikitext('[[' .. talkpage .. '|')
-			:tag(args.mini and 'abbr' or 'span')
-				:attr('title', 'Discuss this template')
-				:cssText(args.fontstyle)
-				:wikitext(args.mini and 'd' or 'talk')
-				:done()
-			:wikitext(']]');
-
-	if not args.noedit then
-		ul
-			:tag('li')
-				:addClass('nv-edit')
-				:wikitext('[' .. title:fullUrl('action=edit') .. ' ')
-				:tag(args.mini and 'abbr' or 'span')
-					:attr('title', 'Edit this template')
-					:cssText(args.fontstyle)
-					:wikitext(args.mini and 'e' or 'edit')
-					:done()
-				:wikitext(']');
 	end
 
-	if args.brackets then
-		div
-			:tag('span')
-				:css('margin-left', '-0.125em')
-				:cssText(args.fontstyle)
-				:wikitext(' &#93;')
+	navBarDiv:node(shortcuts)
+
+	if showBrackets then
+		navBarDiv:node(mw.html.create('span')
+			:css('margin-left', '-0.125em')
+			:cssText(fontStyle)
+			:wikitext(' ' .. _BRACKETS_RIGHT)
+		)
 	end
 
-	if args.collapsible then
-		div
-			:done()
-		:tag('div')
+	navBarDiv = tostring(navBarDiv)
+
+	if isCollapsible then
+		navBarDiv = navBarDiv .. tostring(mw.html.create('div')
 			:css('font-size', '114%')
-			:css('margin', args.mini and '0 4em' or '0 7em')
-			:cssText(args.fontstyle)
-			:wikitext(args[1])
+			:css('margin', isMini and '0 4em' or '0 7em')
+			:cssText(fontStyle)
+			:wikitext(headerText)
+		)
 	end
 
-	return tostring(div:done())
+	return navBarDiv
 end
 
-function p.navbar(frame)
-	if not getArgs then
-		getArgs = require('Module:Arguments').getArgs
+function NavBar._getTitle(titleArg)
+	--if no template title is given via the arguments try to get it
+	if not titleArg then
+		local rootFrame
+		local currentFrame = mw.getCurrentFrame()
+		while currentFrame ~= nil do
+			rootFrame = currentFrame
+			currentFrame = currentFrame:getParent()
+		end
+
+		titleArg = ':' .. rootFrame:getTitle()
```

Comment:
Am I thinking wrong that this would give the page it's used on, not the template?

Reply:
it gives the template
we also use it in infobox implementation
https://github.com/Liquipedia/Lua-Modules/blob/040dd15b0715bba3b9733cb4f6464100f289fd44/components/infobox/commons/infobox_widget_header.lua#L96

Reply:
Ok